scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1386,159 @@ cdef class Rotation:
+            Euler angles specified in radians (`degrees` is False) or degrees
+            (`degrees` is True):
+
+                - array_like with shape (3,), corresponding to single rotation
+                - array_like with shape (N, 3), where each `angle[i, 0]`
+                  corresponds to a single rotation
+
+        degrees : bool, optional
+            If True, then the given angles are assumed to be in degrees. Default is False.
+        extrinsic : boolean
+            If True, sequence will be extrinsic. If False, sequence will be
+            treated as intrinsic.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotation represented by the sequence of
+            rotations around given axes with given angles.
+
+        References
+        ----------
+        .. [1] https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations
+        .. [2] Shuster, Malcolm & Markley, Landis. (2003). Generalization of
+               the Euler Angles. Journal of the Astronautical Sciences. 51. 123-132. 10.1007/BF03546304.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Davenport angles are a generalization of Euler angles, when we use the
+        canonical basis axes:
+
+        >>> ex = [1, 0, 0]
+        >>> ey = [0, 1, 0]
+        >>> ez = [0, 0, 1]
+
+        Initialize a single rotation with a given axis sequence:
+
+        >>> axes = [ez, ey, ex]
+        >>> r = R.from_davenport(axes, [90, 0, 0], degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        It is equivalent to Euler angles in this case:
+
+        >>> r.as_euler('zyx', degrees=True)
+        array([90.,  0., -0.])
+
+        Initialize multiple rotations in one object:
+
+        >>> r = R.from_davenport(axes, [[90, 45, 30], [35, 45, 90]], degrees=True)
+        >>> r.as_quat().shape
+        (2, 4)
+
+        Non-canonical axes are possible, and they do not need to be normalized,
+        as long as consecutive axes are orthogonal:
+
+        >>> e1 = [2, 0, 0]
+        >>> e2 = [0, 1, 0]
+        >>> e3 = [1, 0, 1]
+        >>> axes = [e1, e2, e3]
+        >>> r = R.from_davenport(axes, [90, 45, 30], degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        """
+        num_axes = len(axes)
+        if num_axes != 3:
+            raise ValueError("Expected axes to be list of 3 vectors got {}"
+                             "".format(axes))
+
+        for axis in axes:
+            if len(axis) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        n1 = np.array(axes[0])
+        n2 = np.array(axes[1])
+        n3 = np.array(axes[2])
+
+        if abs(np.dot(n1, n2)) >= 1e-7:
+            raise ValueError("Consecutive axes must be orthogonal.")
+        if abs(np.dot(n2, n3)) >= 1e-7:
+            raise ValueError("Consecutive axes must be orthogonal.")
```

Comment:
Suggest to do these checks after normalization. Possibly relate the threshold to an angle (for better interpretation).

Reply:
Done!